containers/self-hosted-runner/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM ghcr.io/actions/actions-runner:latest AS base
+FROM ghcr.io/actions/actions-runner:latest@sha256:9c3383600872325f21951a0a1a8ae640361ff5e1abdf1f2900de8ba1cfd915e9 AS base
```

Comment:
```suggestion
FROM ghcr.io/actions/actions-runner:2.326.0@sha256:9c3383600872325f21951a0a1a8ae640361ff5e1abdf1f2900de8ba1cfd915e9 AS base
```

Reply:
@mamu0 yep but the actual dockerfile is not a "template", static analysis tools like renovate (or trivy, or future external ones) will fail because the replacement is done on-the-fly outside of docker.

futhermore, the thing that there is a step that replaces the first line have to be documented, remembered etc. 

I think an `ARG` (valued by the workflow) will make this step more explicit and understandable by externals tools without changing this behaviour (the workflow will set the `ARG` instead replacing the line)